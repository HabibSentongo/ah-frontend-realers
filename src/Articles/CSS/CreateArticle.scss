@import '../../commons/assets/variables/font.scss';
@import '../../commons/assets/variables/colors.scss';

.create-article {

  display: flexbox;
  flex-direction: column;
  margin: 20px auto;
  width: 70%;

  .section-title {
    color: $primary-blue;
    font-family: $lato-font;
    font-size: 36px;
    font-weight: 100;
    margin: 30px 0;
    text-align: center;
    width: 100%;
  }

  .status-section {
    margin: 5px 0;
  }

  .form-control {
    border-radius: 0;
  }

  button {
    background-color: $primary-blue !important;
    border-color: lighten($primary-blue, 10%) !important;
    border-radius: 0%;
    color: $white;
    font-family: $source-san-font;
    font-size: 18px;
    font-weight: 100;
    height: 41px;
    width: 212px;
  }

  .call-for-action {
    display: flex;
    justify-content: flex-end;
  }

  .article-photo {
    color: $default-gray;
    font-family: $source-san-font;
    font-size: 16px;
    font-weight: 100;

    .article-photo-name {
      color: $light-gray;
      padding-left: 5px;

      &::before {
        content: ': ';
      }
    }
  }

  .loader {
    margin-bottom: 40px;
  }
}
