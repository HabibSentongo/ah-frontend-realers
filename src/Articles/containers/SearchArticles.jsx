import React, { Component } from 'react';
import { Container } from 'react-bootstrap';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Article from '../../commons/components/Article';
import fetchSearchedArticles from '../redux/actions/SearchArticlesAction';
import '../CSS/SearchArticles.scss';
import ErrorLabel from '../../commons/components/ErrorLabel';
import Loading from '../../commons/components/Loading';

export class SearchArticles extends Component {
  componentDidMount() {
    const { match: { params: { searchkey } }, getSearchedArticles } = this.props;

    getSearchedArticles(searchkey);
  }

  render() {
    const { items, match: { params: { searchkey } }, isSearching } = this.props;
    return (
      <div className="search-articles">
        {
        isSearching === true
          ? <Loading />
          : (
            <Container>
              {items.length !== 0 ? (
                <div>
                  <h2 className="section-title">{`Search Results For - '${searchkey}' `}</h2>
                  <div className="article-display-box">
                    {items.map(article => (
                      <Article key={article.slug} article={article} />
                    ))}
                  </div>
                </div>
              ) : (
                <ErrorLabel
                  message={`Your search - '${searchkey}' did not match any of the articles`}
                  tip="Try again with another word"
                />
              )}
            </Container>
          )
        }
      </div>
    );
  }
}

SearchArticles.propTypes = {
  items: PropTypes.array,
  match: PropTypes.object.isRequired,
  isSearching: PropTypes.bool,
  getSearchedArticles: PropTypes.func.isRequired,
};

SearchArticles.defaultProps = {
  items: [],
  isSearching: false,
};

export const mapStateToProps = state => {
  const { isSearching, items } = state.searchedArticles;
  return { isSearching, items };
};

export const mapDispatchToProps = dispatch => ({
  getSearchedArticles: (searchedKey) => {
    dispatch(fetchSearchedArticles(searchedKey));
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(SearchArticles);
