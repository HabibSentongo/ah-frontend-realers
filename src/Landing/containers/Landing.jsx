/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import '../CSS/Landing.scss';
import RecommendedArticles from '../components/RecommendArticles';
import LatestArticles from '../components/LatestArticles';
import fetchLatestArticles from '../redux/actions/latestArticlesAction';
import Loading from '../../commons/components/Loading';

export class Landing extends Component {
  componentWillMount() {
    this.props.fetchLatestArticles();
  }

  render() {
    const {
      latestArticles,
      recommendedArticles,
      fetchStatus,
      isFetching,
    } = this.props;

    return (
      <div className="landing">
        {
        isFetching === true
          ? <Loading />
          : (
            <div>
              <RecommendedArticles
                recommendedArticles={recommendedArticles}
                fetchStatus={fetchStatus}
              />
              <LatestArticles
                latestArticles={latestArticles}
                fetchStatus={fetchStatus}
              />
            </div>
          )
        }
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  latestArticles: state.latest_articles.items,
  recommendedArticles: state.latest_articles.items,
  fetchStatus: state.latest_articles.status,
  isFetching: state.latest_articles.is_fetching,
});

export default connect(mapStateToProps, { fetchLatestArticles })(Landing);
