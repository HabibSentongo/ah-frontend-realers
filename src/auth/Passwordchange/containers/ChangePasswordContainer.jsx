import { connect } from 'react-redux';
import changePasswordAction from '../redux/actions/postAction';

const mapStateToProps = state => ({
  request_success: state.changePassword.success,
  request_errors: state.changePassword.errors,
  isLoading: state.changePassword.isLoading,
});

export default connect(mapStateToProps, { changePasswordAction });
