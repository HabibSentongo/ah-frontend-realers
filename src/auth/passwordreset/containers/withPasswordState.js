import { connect } from 'react-redux';
import resetPasswordAction from '../redux/actions/postAction';

const mapStateToProps = state => ({
  request_success: state.resetPassword.success,
  request_errors: state.resetPassword.errors,
  isLoading: state.resetPassword.isLoading,
});

export default connect(mapStateToProps, { resetPasswordAction });
