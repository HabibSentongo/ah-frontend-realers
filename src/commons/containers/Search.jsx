import {
  Nav, Form, InputGroup, FormControl, Button,
} from 'react-bootstrap';
import React from 'react';
import '../CSS/Search.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSearch } from '@fortawesome/free-solid-svg-icons';

export class Search extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      searchKey: '',
    };
  }

  handleChange = (event) => {
    const { value } = event.target;
    this.setState({
      searchKey: value,
    });
  }

  render() {
    const { searchKey } = this.state;
    return (
      <Nav className="justify-content-center bg-dark Nav" activeKey="/home">
        <InputGroup className="search-box">
          <InputGroup.Prepend>
            <InputGroup.Text className="search-icon">
              <FontAwesomeIcon icon={faSearch} />
            </InputGroup.Text>
          </InputGroup.Prepend>
          <Form inline onSubmit={(e) => { e.preventDefault(); }}>
            <FormControl
                type="text"
                placeholder="Search articles by tag, title or author ..."
                className="mr-sm-2 search-input"
                onChange={this.handleChange}
                value={searchKey}
                required
              />
            <Button type="button" href={`/search/${searchKey}`} className="search-btn">SEARCH</Button>
          </Form>
        </InputGroup>
      </Nav>
    );
  }
}

export default Search;
