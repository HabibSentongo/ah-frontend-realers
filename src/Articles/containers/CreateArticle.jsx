import {
  Form, Row, Col, Button, ProgressBar,
} from 'react-bootstrap';
import React from 'react';
import '../CSS/CreateArticle.scss';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCamera } from '@fortawesome/free-solid-svg-icons';
import Loader from 'react-loader';
import PropTypes from 'prop-types';
import { storage } from '../../commons/Util/FirebaseConfig';
import postAnArticle from '../redux/actions/CreateArticleAction';
import ErrorLabel from '../../commons/components/ErrorLabel';
import SuccessLabel from '../../commons/components/SuccessLabel';
import { TITLE_REGEX, BODY_REGEX, TAGS_REGEX } from '../constants';

export class CreateArticle extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isFieldInvalid: {
        title: null,
        body: null,
        tags: null,
      },
      articleTitle: '',
      articleBody: '',
      articleTags: '',
      uploadProgress: 0,
      uploadError: 'No Error',
      displayProgress: 'none',
      isFormValid: false,
      articlePhoto: {
        name: 'No Photo selected',
        data: null,
      },
    };
  }

  componentWillMount() {
    const token = localStorage.getItem('user_token');
    if (token === null) {
      window.location = '/login';
    }

    return true;
  }

  handleCreateArticle = (event) => {
    event.preventDefault();
    const {
      articleTitle, articleBody, articleTags, articlePhoto,
    } = this.state;
    const articleData = {
      article: {
        title: articleTitle,
        body: articleBody,
        description: articleBody,
      },
    };

    if (articleTags !== '') {
      const tags = articleTags.split(',');
      articleData.article.tagName = tags;
    }

    const { postAnArticle } = this.props;

    if (articlePhoto.data !== null) {
      this.handleImageUpload();
    } else {
      postAnArticle(articleData);
    }
  }

  handleChange = (event) => {
    const { target: { name, value, files } } = event;
    const { isFieldInvalid } = this.state;

    switch (name) {
      case 'articleTitle':
        isFieldInvalid.title = !TITLE_REGEX.test(value);
        break;
      case 'articleBody':
        isFieldInvalid.body = !BODY_REGEX.test(value);
        break;
      case 'articleTags':
        isFieldInvalid.tags = !TAGS_REGEX.test(value);
        break;
      case 'articlePhoto':
        this.setState({
          articlePhoto: {
            name: 'One Photo Added, click here again to change photo',
            data: files[0],
          },
        });
        break;
      default:
        break;
    }
    if (name !== 'articlePhoto') {
      this.setState({
        [name]: value,
      });
    }

    if ((isFieldInvalid.title === false
        && isFieldInvalid.body === false
    )) {
      this.setState({
        isFormValid: true,
      });
    } else {
      this.setState({
        isFormValid: false,
      });
    }
  }

  handleImageUpload = () => {
    const {
      articleTitle, articleBody, articleTags, articlePhoto,
    } = this.state;

    const { postAnArticle } = this.props;

    this.setState({
      displayProgress: 'block',
    });

    const imageUpLoad = storage.ref(`article-image/${articlePhoto.data.name}`).put(articlePhoto.data);
    imageUpLoad.on('state_changed',
      (snapshot) => {
        const progress = (snapshot.bytesTransferred / snapshot.totalBytes) * 100;
        this.setState({
          uploadProgress: progress,
        });
      },
      (error) => {
        this.setState({
          uploadError: error,
        });
      },
      /* istanbul ignore next */
      (complete) => {
        storage.ref('article-image').child(articlePhoto.data.name).getDownloadURL()
          .then(url => {
            const articleData = {
              article: {
                title: articleTitle,
                body: articleBody,
                description: articleBody,
                image: url,
              },
            };
            if (articleTags !== '') {
              const tags = articleTags.split(',');
              articleData.article.tagName = tags;
            }
            postAnArticle(articleData);
          });
      });
  }

  render() {
    const {
      isFieldInvalid, articlePhoto, articleTitle, articleBody, isFormValid,
      articleTags, uploadProgress, displayProgress, uploadError,
    } = this.state;
    const {
      errorOccured,
      isPosting,
      isArticlePosted,
      createdArticle,
    } = this.props;
    if (createdArticle !== undefined) {
      setTimeout(() => {
        window.location = `/article/${createdArticle.data.slug}`;
      },
      4000);
    }
    return (
      <div className="create-article">
        <h2 className="section-title">Create Article</h2>
        <Form onSubmit={this.handleCreateArticle}>
          <Form.Group controlId="formBasicTitle">
            <Form.Control
              type="text"
              required
              name="articleTitle"
              placeholder="Article Title"
              value={articleTitle}
              onChange={this.handleChange}
              isInvalid={isFieldInvalid.title} />
            {
            isFieldInvalid.title
              ? (
                <Form.Text className="text-muted" variant="danger">
                  Article title can&apos;t be empty or contain only digits or with less than 4 characters
                </Form.Text>
              )
              : (
                <Form.Text className="text-muted" />
              )
            }
          </Form.Group>
          <Form.Group controlId="exampleForm.ControlTextarea1">
            <Form.Control
              name="articleBody"
              value={articleBody}
              onChange={this.handleChange}
              as="textarea"
              rows="15"
              placeholder="Article body"
              isInvalid={isFieldInvalid.body}
              required
            />
            {
            isFieldInvalid.body
              ? (
                <Form.Text className="text-muted">
                  Article body can&apos;t not be empty, should be atleast 100 characters long
                </Form.Text>
              )
              : (
                <Form.Text className="text-muted" />
              )
            }
          </Form.Group>
          <Form.Group controlId="formBasicTag">
            <Form.Control
              name="articleTags"
              value={articleTags}
              onChange={this.handleChange}
              type="text"
              placeholder="Article Tags"
              isInvalid={isFieldInvalid.tags} />
            {
            isFieldInvalid.tags
              ? (
                <Form.Text className="text-muted">
                  Enter tags should only be letters separated by comma like books,education
                </Form.Text>
              )
              : (
                <Form.Text className="text-muted" />
              )
            }
          </Form.Group>
          <Form.Group>
            <Form.Label htmlFor="articlePhoto" style={{ cursor: 'pointer' }}>
              <h5 className="article-photo">
                <span>
                  <FontAwesomeIcon icon={faCamera} />
                  {' '}
                  Add Photo
                </span>
                <span className="article-photo-name">{articlePhoto.name}</span>
              </h5>
              <Form.Control
                    id="articlePhoto"
                    name="articlePhoto"
                    type="file"
                    accept=".png,.jpg"
                    onChange={this.handleChange}
                    style={{ display: 'none' }}
                />
            </Form.Label>
            <Row style={{ display: displayProgress }}>
              {
                uploadError === 'No Error'
                  ? (
                    <Col sm={12}>
                      <ProgressBar now={uploadProgress} label={`${uploadProgress}%`} animated />
                    </Col>
                  )
                  : (
                    <ErrorLabel
                      message={String(uploadError)}
                      tip="try uploading the image again"
                    />
                  )
              }

            </Row>
          </Form.Group>
          <Row className="status-section">
            <Col sm={12}>
              {
                errorOccured !== undefined
                && (
                <ErrorLabel
                message={String(errorOccured.errorMessage.errors.body)}
                tip="edit your article body"
                />
                )
              }
              {
                isArticlePosted && <SuccessLabel message="Article has been successfully created" />
              }
            </Col>
          </Row>
          <Row>
            <Col sm={12} className="call-for-action">
              <Loader loaded={!isPosting} color="#00a4d3">
                <Button type="submit" disabled={!isFormValid} id="submitForm">
               Create Article
                </Button>
              </Loader>
            </Col>
          </Row>
        </Form>
      </div>
    );
  }
}

CreateArticle.protoTypes = {
  postAnArticle: PropTypes.func.isRequired,
  isArticlePosted: PropTypes.bool.isRequired,
};

export const mapStateToProps = state => ({
  isPosting: state.createArticle.isPosting,
  errorOccured: state.createArticle.error,
  isArticlePosted: state.createArticle.status,
  createdArticle: state.createArticle.item,
});

export default connect(mapStateToProps, { postAnArticle })(CreateArticle);
