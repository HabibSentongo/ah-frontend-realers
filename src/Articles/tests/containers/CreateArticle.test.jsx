import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { CreateArticle, mapStateToProps } from '../../containers/CreateArticle';

let mockedState = {
  isFieldInvalid: {
    title: false,
    body: false,
    tags: false,
  },
  articleTitle: 'hello',
  articleBody: 'body',
  articleTags: 'body, boys',
  articlePhoto: {
    name: 'No Photo selected',
    data: null,
  },
};


// test
describe('CreateArticle Component', () => {
  const props = {
    errorOccured: { errorMessage: { errors: { body: [] } } },
    isPosting: false,
    isArticlePosted: false,
    postAnArticle: jest.fn(),
  };
  const wrapper = shallow(<CreateArticle {...props} />);

  it('should match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should text input Change for title', () => {
    const event = {
      target: {
        value: 'New',
        name: 'articleTitle',
      },
    };
    const instance = wrapper.instance();

    instance.handleChange(event);
    expect(instance.state.isFieldInvalid.title).toBe(true);
    expect(instance.state.articleTitle).toBe('New');

    event.target.value = 'New Item';
    instance.handleChange(event);
    expect(instance.state.articleTitle).toBe('New Item');
    expect(instance.state.isFieldInvalid.title).toBe(false);
  });

  it('should image input Change', () => {
    const event = {
      target: {
        value: 'New photo',
        name: 'articlePhoto',
        files: ['image.png'],
      },
    };
    const instance = wrapper.instance();
    instance.handleChange(event);
    expect(instance.state.articlePhoto.name).toBe('One Photo Added, click here again to change photo');
    expect(instance.state.articlePhoto.data).toBe('image.png');
  });

  it('should test article body input change', () => {
    const event = {
      target: {
        value: 'I am the new body which is invalid',
        name: 'articleBody',
      },
    };
    const instance = wrapper.instance();

    instance.handleChange(event);
    expect(instance.state.isFieldInvalid.body).toBe(true);

    event.target.value = 'I am the new body which is valid I am the new body which is valid I am the new body which is valid I am the new body which is invalid I am the new body which is invalid';
    instance.handleChange(event);
    expect(instance.state.isFieldInvalid.title).toBe(false);
  });

  it('should test article tags input change', () => {
    const event = {
      target: {
        value: '2',
        name: 'articleTags',
      },
    };
    const instance = wrapper.instance();

    instance.handleChange(event);
    expect(instance.state.isFieldInvalid.tags).toBe(true);

    event.target.value = 'kampala,uganda';
    instance.handleChange(event);
    expect(instance.state.isFieldInvalid.tags).toBe(false);
  });

  it('should return false when unkown field name is passed', () => {
    const event = {
      target: {
        value: '2',
        name: 'hfhgfg',
      },
    };
    const instance = wrapper.instance();

    instance.handleChange(event);
    expect(instance.state.isFieldInvalid.tags).toBe(false);
  });

  it('should set isFormValid to true/false if all fields are true/false', () => {
    const event = {
      target: {
        value: 'kampala',
        name: 'articleTags',
      },
    };
    const instance = wrapper.instance();
    instance.setState(mockedState);
    instance.handleChange(event);
    expect(instance.state.isFormValid).toBe(true);

    // event.target.value = '';
    // instance.handleChange(event);
    // console.log('instance.state',event.target.value);
    // // expect(instance.state.isFormValid).toBe(false);
  });

  it('on submit fn is called upon form submit', () => {
    const event = {
      preventDefault: jest.fn(),
    };
    const state = {
      articleTitle: 'hello',
      articleBody: 'I am the new body for you',
      articleTags: 'body, boys',
      articlePhoto: {
        data: null,
        name: 'No Image',
      },
    };
    const instance = wrapper.instance();
    instance.setState(state);
    instance.handleCreateArticle(event);

    expect(props.postAnArticle).toHaveBeenCalled();
  });

  it('should call upload image when form has an image', () => {
    const event = {
      preventDefault: jest.fn(),
    };
    const state = {
      isFieldInvalid: { title: false, body: false, tags: false },
      articleTitle: 'hello',
      articleBody: 'I am the new body for you',
      articleTags: 'body, boys',
      articlePhoto: {
        data: new File(['logo'], 'new-logo.png'),
        name: 'new-logo.png',
      },
    };
    const instance = wrapper.instance();
    instance.setState(state);
    instance.handleCreateArticle(event);
    expect(instance.state.articlePhoto.name).toBe('new-logo.png');
  });

  it('should handle form submit with image upload', () => {
    const state = {
      articleTitle: 'hello',
      articleBody: 'I am the new body for you',
      articleTags: 'body, boys',
      articlePhoto: {
        data: new File(['image data'], 'logo.png'),
        name: 'logo.png',
      },
    };
    const instance = wrapper.instance();
    instance.setState(state);
    instance.handleImageUpload();
    expect(instance.state.articlePhoto.name).toBe('logo.png');
  });
});

describe('mapStateToProps', () => {
  it('testing mapStateToProps', () => {
    mockedState = {
      createArticle: {
        status: true,
        isPosting: true,
      },
    };
    const state = mapStateToProps(mockedState);
    expect(state.isPosting).toBe(true);
  });
});
