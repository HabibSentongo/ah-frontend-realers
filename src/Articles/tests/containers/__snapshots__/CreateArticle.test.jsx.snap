// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateArticle Component should match the snapshot 1`] = `
<div
  className="create-article"
>
  <h2
    className="section-title"
  >
    Create Article
  </h2>
  <Form
    inline={false}
    onSubmit={[Function]}
  >
    <FormGroup
      controlId="formBasicTitle"
    >
      <FormControl
        isInvalid={null}
        name="articleTitle"
        onChange={[Function]}
        placeholder="Article Title"
        required={true}
        type="text"
        value=""
      />
      <FormText
        className="text-muted"
      />
    </FormGroup>
    <FormGroup
      controlId="exampleForm.ControlTextarea1"
    >
      <FormControl
        as="textarea"
        isInvalid={null}
        name="articleBody"
        onChange={[Function]}
        placeholder="Article body"
        required={true}
        rows="15"
        value=""
      />
      <FormText
        className="text-muted"
      />
    </FormGroup>
    <FormGroup
      controlId="formBasicTag"
    >
      <FormControl
        isInvalid={null}
        name="articleTags"
        onChange={[Function]}
        placeholder="Article Tags"
        type="text"
        value=""
      />
      <FormText
        className="text-muted"
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        htmlFor="articlePhoto"
        srOnly={false}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        <h5
          className="article-photo"
        >
          <span>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f030",
                    "M512 144v288c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V144c0-26.5 21.5-48 48-48h88l12.3-32.9c7-18.7 24.9-31.1 44.9-31.1h125.5c20 0 37.9 12.4 44.9 31.1L376 96h88c26.5 0 48 21.5 48 48zM376 288c0-66.2-53.8-120-120-120s-120 53.8-120 120 53.8 120 120 120 120-53.8 120-120zm-32 0c0 48.5-39.5 88-88 88s-88-39.5-88-88 39.5-88 88-88 88 39.5 88 88z",
                  ],
                  "iconName": "camera",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
             
            Add Photo
          </span>
          <span
            className="article-photo-name"
          >
            No Photo selected
          </span>
        </h5>
        <FormControl
          accept=".png,.jpg"
          id="articlePhoto"
          name="articlePhoto"
          onChange={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="file"
        />
      </FormLabel>
      <Bootstrap(Row)
        style={
          Object {
            "display": "none",
          }
        }
      >
        <Col
          sm={12}
        >
          <Bootstrap(ProgressBar)
            animated={true}
            label="0%"
            now={0}
          />
        </Col>
      </Bootstrap(Row)>
    </FormGroup>
    <Bootstrap(Row)
      className="status-section"
    >
      <Col
        sm={12}
      >
        <ErrorLabel
          message=""
          tip="edit your article body"
        />
      </Col>
    </Bootstrap(Row)>
    <Bootstrap(Row)>
      <Col
        className="call-for-action"
        sm={12}
      >
        <Component
          color="#00a4d3"
          component="div"
          loaded={true}
          loadedClassName="loadedContent"
          parentClassName="loader"
        >
          <Button
            active={false}
            disabled={true}
            id="submitForm"
            type="submit"
            variant="primary"
          >
            Create Article
          </Button>
        </Component>
      </Col>
    </Bootstrap(Row)>
  </Form>
</div>
`;
